### DISCUSSION TITLE
Celery fills Redis Broker with results without setting TTL - resulting in OOM of Redis

### DISCUSSION THREAD

**Discussion by Flauschbaellchen** [celery/celery#7776] [Q&A]
I have an issue in our application: When running Celery tasks, Celery keeps track of the tasks saving metadata (and the task's result) into the Broker's database. This data is however never deleted, which results in OOM of Redis eventually.

### Details
I'm using the Redis backend. After running 1-3 tasks, the Redis database consists of the following entries:

```
127.0.0.1:6379> keys *
1) "_kombu.binding.w3syse.queues.default"
2) "_kombu.binding.celery.pidbox"
3) "7a31e86c-24fe-3722-8f2f-997e0cbf9786"
4) "_kombu.binding.celeryev"
```
(I've set `task_default_queue = "w3syse.queues.default"`)

Celery created the additional key `7a31e86c-24fe-3722-8f2f-997e0cbf9786` which tracks a task.
The content is like the following:
```
{"task_id": "9130013f-cabe-4ae2-9c44-8dc78272526b", "status": "SUCCESS", "result": null, "traceback": null, "children": []}
{"task_id": "9130013f-cabe-4ae2-9c44-8dc78272526b", "status": "STARTED", "result": {"pid": 1, "hostname": "celery@worker-server-7889c959d7-5mcmp"}, "traceback": null, "children": []}
```

The issue is, that those keys are never deleted, spamming the whole Redis database to such an extend, that the size is greater than the RAM of the machine and Redis is not able to run anymore (64GB+).
This happens regularly if running several hundreds of tasks per second.

After some research, I've found that the RedisBackend should be responsible of deleting such keys automatically on its own: https://github.com/celery/celery/blob/4f54119b822374533934943fa41ba68011295489/celery/backends/redis.py#L203

The TTL is 1 day (default) and can be adjusted by setting `result_expires` (see: https://docs.celeryq.dev/en/stable/userguide/configuration.html#result-expires).

I tried to set it to `5` (5 seconds), however, when checking the TTL of the key, Redis always returns `-1`, which means "No expire":
```
127.0.0.1:6379> ttl 7a31e86c-24fe-3722-8f2f-997e0cbf9786
(integer) -1
```

### Environment

Versions of packages:
* celery[redis]==5.2.7
* django-celery-beat==2.3.0
* kombu==5.2.4
* redis==3.5.3

Redis server is 7.0.4.

Other settings:
```
CELERY_RESULT_BACKEND = "rpc://"
CELERY_ACCEPT_CONTENT = ["application/json"]
CELERY_TASK_SERIALIZER = "json"
CELERY_RESULT_SERIALIZER = "json"
CELERY_TIMEZONE = TIME_ZONE
CELERY_ENABLE_UTC = True
CELERY_BEAT_SCHEDULER = "django_celery_beat.schedulers:DatabaseScheduler"
CELERY_TASK_SEND_SENT_EVENT = True
CELERY_WORKER_SEND_TASK_EVENTS = True
CELERY_TASK_TRACK_STARTED = True
```

Can someone reproduce this issue/bug or can give me a hint if I need to adjust a setting? - Thanks a lot!

**Comment by auvipy** [comment]
can you try more latest version of redis? https://github.com/redis/redis-py/releases/tag/v4.3.4

**Reply by Flauschbaellchen** [reply]
Hi @auvipy, I've tried with the newest version and the issue is the same. The TTL on the redis side is still unset (-1).
